bcgov/cas-registration
# Responsive MUI Header, Content, Footer components

Addresses [Ticket 112](https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/112)

## 🚧 Changes
    - Add client\app\component\theme\EmotionCache
    - Update client\app\component\layout\Footer
        - Add responsive props 
    - Update client\app\component\layout\Header
        - Add responsive props 
    - Update client\app\page.tsx
        - Add responsive props 
    - Update client\app\layout.tsx
        - Add responsive viewport meta tag 
    - Update client\tailwind.config.js

## 🚀 Impact:
   - Implemented Emotion cache to make style configuration and theme available to all components (SSR & CSR) in the component tree.
   - Configured tailwind.config to combining the utility-first approach of Tailwind CSS with the rich component library of Material-UI with: ` important: "#__next", `
   - Viewport meta tag to ensure proper rendering and touch zooming for all devices
   - Refined layout header\footer components elements as per Figma design for mobile and desktop using MUI theme breakpoints
   - Place holder page for app landing page using MUI grid for responsive, ordered layout

## 📝 Notes:
- Header Log In links are place holders see [ticket 14](https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/14)
- Page content is place holder see [ticket 163](https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/163)
- Footer Home\Contact Us links are place holders
- **Optional:** It is best practice while using Tailwind and MUI together to add a prefix in the tailwind class and by setting tailwind to important so that class conflicts are reduced between these two libraries.
```
// tailwind.config.js
module.exports = {
  prefix: 'tw-',   👈 Use your desired prefix 
}
```

##  🔬 Testing:

1. Modify .env file as per .env.example
2. Open terminal in cas-registration\client folder
3. Start the development server:
   ```
   yarn dev
   ```
4. In your web browser, navigate to `http://localhost:3000/`
5. You should see the "place holder" content for [Workflow | Request Admin Access to Org from CG - 1](https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=50-3570&mode=design&t=P0FJ8ggBAmCxrwQ0-0)
6. Using the Dev Tools\Toggle device toolbar set the screen width to mimic Figma screen (width: 1440px)
![image](https://github.com/bcgov/cas-registration/assets/120038448/edde2c01-01b4-4c70-b791-1ef4f9e6f09c)
7. Validate the Header elements align with Figma screen
8. Validate the Footer elements align with Figma screen
9. Using the Dev Tools\Toggle device toolbar set the screen width to width< 900px
![image](https://github.com/bcgov/cas-registration/assets/120038448/751feb7c-0dab-46cf-acae-8d5bdd4eb094)
10. Login in buttons are below Logo
11. The Content "Key Dates" column is ordered first.
12.  Using the Dev Tools\Toggle device toolbar set the screen width to width< 600px
13. Footer links are centred vertical
![image](https://github.com/bcgov/cas-registration/assets/120038448/2520bbfc-4dbe-42d4-90e3-10a44b7bdb73)

**Post Branch Merge**
1.  [Prod url](https://cas-reg-frontend-dev.apps.silver.devops.gov.bc.ca/) displays components as above...
2. Notify devs of client\.env changes
